### ISSUE TITLE
switching sentry-relay mode doens't work as expected.

### ISSUE THREAD

**Issue by prasus** [sentry-kubernetes/charts#403]
Coming here from [sentry/issues/22789](https://github.com/getsentry/sentry/issues/22789) 🙂 

I have got an on-premise sentry setup deployed with sentry/sentry chart version 8.1.2 and I am experiencing the issue explained at [sentry/issues/22789](https://github.com/getsentry/sentry/issues/22789), The errors events are always accepted and and also the rate-limit settings are not honored.

The fix for this is available in the [latest chart](https://github.com/sentry-kubernetes/charts/pull/353) and it's basically switching the sentry-relay mode from `proxy` to `managed` 

I have redeployed the my current setup (based on chart version 8.1.2) after making the following change,

From:

relay:
  mode: proxy

To:

relay:
  mode: managed

Ideally this should work, but the sentry relay pod fails with the below error after the change and I had to rollback the changes to make it work again.

2021-06-07T04:13:17Z [relay_server::actors::upstream] INFO: registering with upstream (http://sentry-web:9000/)
2021-06-07T04:13:17Z [relay_server::actors::upstream] ERROR: authentication encountered error: could not send request to upstream
  caused by: could not send request using actix-web client
  caused by: Server disconnected
  caused by: Server disconnected
2021-06-07T04:13:34Z [relay_server::actors::upstream] INFO: registering with upstream (http://sentry-web:9000/)
2021-06-07T04:13:34Z [relay_server::actors::upstream] ERROR: authentication encountered error: could not send request to upstream
  caused by: could not send request using actix-web client
  caused by: Server disconnected
  caused by: Server disconnected

Below is our config (taken from the pod when running with mode: proxy)

```
relay:
  mode: proxy
  upstream: "http://sentry-web:9000/"
  host: 0.0.0.0
  port: 3000

processing:
  enabled: true

  kafka_config:
    - name: "bootstrap.servers"
      value: kafka:9092
    - name: "message.max.bytes"
      value: 50000000  # 50MB or bust
  redis: "redis://:@redis-sentry-redis-master:6379"
```
Any help on fixing this would be much appreciated. Thanks!

**Comment by therealtbs** [comment]
I fixed this in my install by adding the following to the `SENTRY_WEB_OPTIONS` in the `sentry.conf.py`:

```diff
 SENTRY_WEB_OPTIONS = {
     "http": "%s:%s" % (SENTRY_WEB_HOST, SENTRY_WEB_PORT),
     "protocol": "uwsgi",
     # This is needed to prevent https://git.io/fj7Lw
     "uwsgi-socket": None,
     "http-keepalive": True,
+    "http-auto-chunked": True,
     "memory-report": False,
     # 'workers': 3,  # the number of web workers
 }
```

In the Helm `values.yaml` it should work like this (I haven't tested this yet):

```yaml
config:
  sentryConfPy: |
    SENTRY_WEB_OPTIONS["http-auto-chunked"] = True
```


I haven't yet figured out if this causes any issues down the line, but it's at least letting me start all pods. I currently don't have any meaningful traffic on this instance so I'm afraid I can't be of much help testing this configuration.

This was inspired by getsentry/onpremise#499. It seems like `http-auto-chunked` is the only necessary part, but it might not hurt to add the rest of the options that were added in that PR

**Comment by Mokto** [comment]
Thanks for this! I also had this issue (but rarely).

I made a try with those options here : https://github.com/sentry-kubernetes/charts/pull/406

I'll try it for a few days.

**Comment by Mokto** [comment]
Just merged it you can try 10.3.3
